samcli/local/docker/lambda_build_container.py

```diff
@@ -0,0 +1,132 @@
+"""
+Represents Lambda Build Containers.
+"""
+
+import json
+import os
+import logging
+
+from .container import Container
+
+LOG = logging.getLogger(__name__)
+
+
+class LambdaBuildContainer(Container):
+    """
+    Class to manage Build containers that are capable of building AWS Lambda functions.
+    This container mounts necessary folders, issues a command to the Lambda Builder CLI,
+    and if the build was successful, copies back artifacts to the host filesystem
+    """
+
+    _IMAGE_REPO_NAME = "lambci/lambda"
+    _BUILDERS_EXECUTABLE = "lambda-builders"
+
+    def __init__(self,
+                 language,
+                 dependency_manager,
+                 application_framework,
+                 source_dir,
+                 manifest_path,
+                 runtime,
+                 optimizations=None,
+                 options=None,
+                 log_level=False):
+
+        manifest_file_name = os.path.basename(manifest_path)
+        manifest_dir = os.path.dirname(manifest_path)
+
+        container_dirs = LambdaBuildContainer._get_container_dirs(source_dir, manifest_dir)
+
+        request_json = self._make_request(language,
+                                          dependency_manager,
+                                          application_framework,
+                                          container_dirs,
+                                          manifest_file_name,
+                                          runtime,
+                                          optimizations,
+                                          options)
+
+        image = LambdaBuildContainer._get_image(runtime)
+        entry = LambdaBuildContainer._get_entrypoint(request_json)
+        cmd = []
+
+        additional_volumes = {
+            # Manifest is mounted separately in order to support the case where manifest
+            # is outside of source directory
+            manifest_dir: {
+                "bind": container_dirs["manifest_dir"],
+                "mode": "ro"
+            }
+        }
+
+        env_vars = None
+        if log_level:
+            env_vars = {
+                "LAMBDA_BUILDERS_LOG_LEVEL": log_level
+            }
+
+        super(LambdaBuildContainer, self).__init__(
+            image,
+            cmd,
+            container_dirs["source_dir"],
+            source_dir,
+            additional_volumes=additional_volumes,
+            entrypoint=entry,
+            env_vars=env_vars)
+
+    @staticmethod
+    def _make_request(language,
+                      dependency_manager,
+                      application_framework,
+                      container_dirs,
+                      manifest_file_name,
+                      runtime,
+                      optimizations,
+                      options):
+
+        return json.dumps({
```

Comment:
We should be using OrderedDicts instead. Probably ok but we have been burnt by py2 and py3 differences in the past.

Reply:
I think this should be fine, in this particular case.

Reply:
Probably, but that actually sets the wrong expectation. JSON dicts are expected to be unordered. So I think it is better not using ordered dict here.